x/hub/keeper/keeper.go

```diff
@@ -0,0 +1,98 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ibctypes "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint/types"
+
+	"github.com/dymensionxyz/dymension-rdk/x/hub/types"
+)
+
+type Keeper struct {
+	cdc           codec.BinaryCodec
+	storeKey      storetypes.StoreKey
+	channelKeeper types.ChannelKeeper
+}
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeKey storetypes.StoreKey,
+	channelKeeper types.ChannelKeeper,
+) Keeper {
+	return Keeper{
+		cdc:           cdc,
+		storeKey:      storeKey,
+		channelKeeper: channelKeeper,
+	}
+}
+
+// ExtractHubFromChannel extracts the hub from the IBC port and channel.
+// Returns nil if the hub is not found.
+func (k Keeper) ExtractHubFromChannel(
+	ctx sdk.Context,
+	hubPortOnRollapp string,
+	hubChannelOnRollapp string,
+) (*types.Hub, error) {
+	// Check if the packet is destined for a hub
+	chainID, err := k.ExtractChainIDFromChannel(ctx, hubPortOnRollapp, hubChannelOnRollapp)
+	if err != nil {
+		return nil, err
+	}
+	k.SetHub(ctx, types.Hub{
+		Id:               chainID,
+		ChannelId:        hubChannelOnRollapp,
+		RegisteredDenoms: nil,
+	})
```

Comment:
not really correct because how do you know this is the hub?
and why are you resetting denoms to nil?